src/backend/aspen/api/views/samples.py

```diff
@@ -211,3 +224,63 @@ async def delete_sample(
     await db.delete(sample)
     await db.commit()
     return SampleDeleteResponse(id=sample_db_id)
+
+
+@router.put("/")
+async def update_samples(
+    update_samples_request: UpdateSamplesRequest,
+    db: AsyncSession = Depends(get_db),
+    settings: Settings = Depends(get_settings),
+    user: User = Depends(get_auth_user),
+) -> SamplesResponseSchema:
+
+    # reorganize request data to make it easier to update
+    reorganized_request_data = {s.id: s for s in update_samples_request.samples}
+    sample_ids_to_update = list(reorganized_request_data.keys())
+
+    # Make sure these samples exist and are delete-able by the current user.
+    sample_db_res = await get_owned_samples_by_ids(db, sample_ids_to_update, user)
+    editable_samples = sample_db_res.all()
+
+    # are there any samples that can't be updated?
+    uneditable_samples = [
+        s for s in sample_ids_to_update if s not in [i.id for i in editable_samples]
+    ]
+    if uneditable_samples:
+        raise ex.NotFoundException("some samples cannot be updated")
+    # Check that we don't have dupe public/private ID's
+    [sample.private_identifier for sample in editable_samples]
+    [sample.public_identifier for sample in editable_samples]
```

Comment:
I think you intended to flesh this out, yeah?

Reply:
That's leftover from me trying to generate nicer duplicate-name error messages, which didn't go well and got nixed. Removed!